src/CalcViewModel/StandardCalculatorViewModel.cpp

```diff
@@ -1497,12 +1469,9 @@ void StandardCalculatorViewModel::Recalculate(bool fromHistory)
 
 CommandType StandardCalculatorViewModel::GetSelectedTokenType(_In_ unsigned int tokenPosition)
 {
-    pair<wstring, int> token;
-    shared_ptr<IExpressionCommand> tokenCommand;
-    IFTPlatformException(m_tokens->GetAt(tokenPosition, &token));
-
+    const pair<wstring, int>& token = m_tokens->at(tokenPosition);
     unsigned int tokenCommandIndex = token.second;
-    IFTPlatformException(m_commands->GetAt(tokenCommandIndex, &tokenCommand));
+    const shared_ptr<IExpressionCommand>& tokenCommand = m_commands->at(tokenCommandIndex);
```

Comment:
We can probably remove `tokenCommandIndex` and directly use token.second.

Reply:
I'm cool with that change but I wanted to point out that tokenCommandIndex is also used on line 1348.  I wasn't sure if you all would like to keep better named variable around or to use `token.second` in both places.

Reply:
you can use token.second in both cases. Both are correct (and the compiler will generate the same code), so do what you like best!

Reply:
Cool, I'll do that.  I was just thinking for future code, it might not be a bad idea to used C++17's structured binding with these pairs.  I believe you can do that within the range for loops and have better names than `token.first` and `token.second`.